From: Steven Newbury <steve@snewbury.org.uk>
Subject: [PATCH] Allow buiding without microphone suppport
Date: Sun, 19 Feb 2017 08:01:00 UTC

Unconditional microphone support means portaudio isn't optional.
Make microphone optional to restore the possibility of building
without the extra dependency on portaudio.
---

--- ./CMakeLists.txt.orig	2017-01-16 10:09:20.336834372 +0000
+++ ./CMakeLists.txt	2017-01-16 10:10:38.167338460 +0000
@@ -15,6 +15,7 @@
 option(ENABLE_LTO "Enables Link Time Optimization" OFF)
 option(ENABLE_GENERIC "Enables generic build that should run on any little-endian host" OFF)
 option(ENABLE_HEADLESS "Enables running Dolphin as a headless variant" OFF)
+option(ENABLE_MICROPHONE "Enables Wii microphone support" ON)
 option(ENABLE_ALSA "Enables ALSA sound backend" ON)
 option(ENABLE_AO "Enables libao sound backend" ON)
 option(ENABLE_PULSEAUDIO "Enables PulseAudio sound backend" ON)
@@ -571,7 +572,7 @@
 
 endif()
 
-if(NOT ANDROID)
+if(NOT ANDROID AND ENABLE_MICROPHONE)
 	set(PORTAUDIO_FOUND TRUE)
 	add_definitions(-DHAVE_PORTAUDIO=1)
 
